tests/core/providers/test_tester_provider.py

```diff
@@ -62,3 +62,29 @@ def test_eth_tester_provider_properly_handles_eth_tester_error_messages(
         TransactionFailed, match="execution reverted: The error message."
     ):
         provider.make_request(RPCEndpoint("eth_blockNumber"), [])
+
+
+def test_eth_tester_provider_properly_handles_eth_tester_key_error_messages():
+    provider = EthereumTesterProvider(api_endpoints={})
+    response = provider.make_request(RPCEndpoint("eth_blockNumber"), [])
```

Comment:
Does this raise? If it does, it might be nice to make sure the exception class is what we expect, similar to how we do in the test above with a `with pytest.raises: ...`

Reply:
For missing or not implemented errors, it will not raise but instead return a reponse with an RPCError